.github/workflows/sign.yaml

```diff
@@ -3,58 +3,52 @@ name: Test
 # only trigger on pull request closed events
 on:
   pull_request:
-    types: [ closed ]
+    branches:
+      - main
 
 jobs:
   merge_job:
-    # this job will only run if the PR has been merged
-    if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
     permissions:
       contents: write
       actions: read
       id-token: write # Needed for OIDC and keyless signing
     steps:
-    - run: |
-        echo PR #${{ github.event.number }} has been merged, beginning signing flow
     - name: Checkout
       uses: actions/checkout@v3
       with:
+        ref: ${{ github.head_ref }}
         fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+    - name: Get changed files in the manifests folder
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+      with:
+        files: manifests/*.{yaml,yml}
+    - name: List all changed files
```

Comment:
What I was thinking was you don't want anyone to modify an unsigned PE, you always want them to create a new one. So we should only look for _new_ files in that case. And this is why in my action I was trying to clear the `/unsigned` dir after each successfully merged PR to empty out the dropbox.

Reply:
How are you going to enforce this ?